bokeh/sphinxext/bokeh_plot.py

```diff
@@ -241,26 +238,35 @@ def process_source(self, source, path, js_filename):
 
         return (script_tag, js_path, source, docstring, height_hint)
 
-    def process_sampledata(self, targetnode, source):
-
-        if not hasattr(self.env, 'all_sampledata_xrefs'):
-            self.env.all_sampledata_xrefs = []
-
-        regex = "(:|bokeh\.)sampledata(:|\.| import )\s*(\w+(\,\s*\w+)*)"
-        matches = re.findall(regex, source)
-        if matches:
-            keywords = set()
-            for m in matches:
-                keywords.update(m[2].replace(" ","").split(','))
-            for keyword in keywords:
-                xref_node = sampledata_node('')
-                self.state.nested_parse(self.content, self.content_offset, xref_node)
-
-                self.env.all_sampledata_xrefs.append({
-                    'docname': self.env.docname,
-                    'target': targetnode,
-                    'keyword': keyword
-                })
+    def process_sampledata(self, source):
+
+        if not hasattr(self.env, 'solved_sampledata'):
+            self.env.solved_sampledata = []
+
+        file, lineno =  self.get_source_info()
+        # collect links to all standalone examples
+        if '/docs/examples/' in file and not file in self.env.solved_sampledata:
+            self.env.solved_sampledata.append(file)
```

Comment:
I hope that I can avoid duplicated entries using this list. While talking aboout it, how can I build the documentation in parallel mode?

Reply:
I add `-j 16` on OSX but I am not sure whether parallel build is supported on all platforms (it may be I just don't know)